docs/src/extend/custom-rules.md

```diff
@@ -688,22 +746,30 @@ For examples of using `SourceCode#getScope()` to track variables, refer to the s
 
 ### Accessing Code Paths
 
-ESLint analyzes code paths while traversing AST.
-You can access that code path objects with five events related to code paths.
+ESLint analyzes code paths while traversing AST. You can access code path objects with five events related to code paths. For more information, refer to [Code Path Analysis](code-path-analysis).
 
-[details here](code-path-analysis)
+### Deprecated `SourceCode` Methods
+
+Please note that the following `SourceCode` methods have been deprecated and will be removed in a future version of ESLint:
+
+* `getComments()`: Replaced by `SourceCode#getCommentsBefore()`, `SourceCode#getCommentsAfter()`, and `SourceCode#getCommentsInside()`.
+* `getTokenOrCommentBefore()`: Replaced by `SourceCode#getTokenBefore()` with the `{ includeComments: true }` option.
+* `getTokenOrCommentAfter()`: Replaced by `SourceCode#getTokenAfter()` with the `{ includeComments: true }` option.
+* `isSpaceBetweenTokens()`: Replaced by `SourceCode#isSpaceBetween()`
+* `getJSDocComment()`
+* `parserOptions`: The parser options configured for this run (more details [here](../use/configure/language-options#specifying-parser-options)).
```

Comment:
`parserOptions` are not on `SourceCode` objects but on `context` objects.

Reply:
```suggestion
### Deprecated `SourceCode` Methods

Please note that the following `SourceCode` methods have been deprecated and will be removed in a future version of ESLint:

* `getComments()`: Replaced by `SourceCode#getCommentsBefore()`, `SourceCode#getCommentsAfter()`, and `SourceCode#getCommentsInside()`.
* `getTokenOrCommentBefore()`: Replaced by `SourceCode#getTokenBefore()` with the `{ includeComments: true }` option.
* `getTokenOrCommentAfter()`: Replaced by `SourceCode#getTokenAfter()` with the `{ includeComments: true }` option.
* `isSpaceBetweenTokens()`: Replaced by `SourceCode#isSpaceBetween()`
* `getJSDocComment()`
```

Reply:
ok removing here and adding to context object section